mesa/model.py

```diff
@@ -77,6 +77,20 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._steps: int = 0
         self._time: TimeT = 0  # the model's clock
 
+        # Wrap the user-defined step method
+        if hasattr(self, "step"):
```

Comment:
What if the user defines their step with a different name, e.g. in the simultaneous activation?

Reply:
Good edge case! I think when using a scheduler, you're also mainly using the scheduler's time, so `model.schedule.time`. And as long as one of the functions is a `step()`, the model time will also be increased.

Personally I'm in favor of moving away from the schedulers, but this PR won't break that.

Reply:
> If the model doesn't have a step, it is more informative to raise an error with an informative message.

If you want to use your own process without using Mesa's internal clock, it's perfectly valid. I think we can cover this with documentation, and this will be less and less a problem as we move away from schedulers.

I added an example of staged time increase to the first post.

> Alternatively, bumping the step & time whenever model.agents.do is called, is another alternative.

The problem is, how do you handle multiple agents.do calls? Or agents.do to a partial agentset? Or with a select in between?

Reply:
> > Alternatively, bumping the step & time whenever model.agents.do is called, is another alternative.

> The problem is, how do you handle multiple agents.do calls? Or agents.do to a partial agentset? Or with a select in between?

If we want to track the number of steps, it should be tied to `model.step`. It's a bad idea to tie it to the AgentSet. Conceptually, this would mix up two completely different things. Which, as @EwoutH indicates, gives rise to all kinds of knock-on problems.

Reply:
> If you want to use your own process without using Mesa's internal clock, it's perfectly valid. I think we can cover this with documentation, and this will be less and less a problem as we move away from schedulers.

`steps`'s need is hardcoded it the batch runner and visualization. Currently, they are not optional.